src/views/RoofSelectionView.vue

```diff
@@ -1,19 +1,82 @@
 <script setup lang="ts">
 import { useViewsStore } from '@/stores/views'
-import { onMounted } from 'vue'
+import { computed, inject, onMounted, ref } from 'vue'
 import { viewList } from '@/model/views.model'
 import { usePanelsStore } from '@/stores/panels'
+import FigureEnergy from '@/components/roof_selection/FigureEnergy.vue'
+import AddressTooltip from '@/components/roof_selection/AddressTooltip.vue'
+import TextDidYouKnow from '@/components/roof_selection/TextDidYouKnow.vue'
+import type { RennesApp } from '@/services/RennesApp'
+import RoofSelectionTooltip from '@/components/roof_selection/RoofSelectionTooltip.vue'
+import { useAddressStore } from '@/stores/address'
+import {
+  cleanHighlightedRoofs,
+  disableSelectRoofInteraction,
+  enableSelectRoofInteraction,
+} from '@/services/interactionUtils'
 
-const viewStore = useViewsStore()
+const rennesApp = inject('rennesApp') as RennesApp
 const panelsStore = usePanelsStore()
+const viewStore = useViewsStore()
+const addressStore = useAddressStore()
+let isOpen = ref(true)
+let isBuildingSelectionActive = ref(false)
+let addressClosedByUser = ref(false)
+
+let DISTANCE_MAX_FOR_SELECTION = 600
 
 onMounted(() => {
   viewStore.setCurrentView(viewList['roof-selection'])
-  panelsStore.setTypePanelDisplay('left')
+  panelsStore.setTypePanelDisplay('float-left')
   panelsStore.isCompletelyHidden = true
+  rennesApp
+    .get3DMap()
+    .getScene()
+    .postRender.addEventListener(() => {
```

Comment:
postRender in 3D is called continuously. Even if you don't move the post render map is called.
To avoid this, save the viewpoint and compare the viewpoint with the new one. And if they are not different, do not make the post render content.
You two exemples in trambus:
https://github.com/sigrennesmetropole/cooperation_jn_trambus/blob/main/src/components/map/CompassComponent.vue  , line 18
https://github.com/sigrennesmetropole/cooperation_jn_trambus/blob/main/src/services/aboveMapService.ts , line 27

Reply:
When the user doesn't move the code don't go in if or else if